src/preamble.js

```diff
@@ -2509,7 +2509,7 @@ function integrateWasmJS(Module) {
       } else if (curr === 'interpret-asm2wasm' || curr === 'interpret-s-expr' || curr === 'interpret-binary') {
         if (exports = doWasmPolyfill(global, env, providedBuffer, curr)) break;
       } else {
-        throw 'bad method: ' + curr;
+        Module['quit'](1, 'bad method: ' + curr);
```

Comment:
I think calling `quit` here is not the best thing. `quit` is the lowest layer here, and by calling it, you are forced to add logic into `shell.js` which is higher than it should be. Instead, how about calling `abort` here, and in `abort` (in `postamble.js`), checking if `onRuntimeInitializeFailed` exists and if so, call it?